pages/profile/[id].tsx

```diff
@@ -0,0 +1,114 @@
+import { GetServerSideProps, NextPage } from 'next';
+import absoluteUrl from 'next-absolute-url';
+import React from 'react';
+
+import { UserProfile } from 'server/types/User';
+
+type Props = {
+  user?: UserProfile;
+};
+
+const ProfilePage: NextPage<Props> = ({ user }: Props) => {
+  function possessiveForm(name: string): string {
+    return name.endsWith('s') ? `${name}'` : `${name}'s`;
+  }
+
+  return (
+    <>
+      <Container className="flex flex-col gap-4 pt-20 items-center">
+        <Box component="header">
+          <Typography variant="h1" className="text-5xl font-700 text-center">
+            {possessiveForm(user.discord_name)} Profile
+          </Typography>
+        </Box>
+
+        {user.bio && <Typography variant="body1" component="p">{user.bio}</Typography>}
+
+        <List className="flex gap-2">
+          {user.twitter_username && (
+            <ListItemButton
+              component="a"
+              href={`https://twitter.com/${user.twitter_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <TwitterIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.linkedin_url && (
+            <ListItemButton
+              component="a"
+              href={user.linkedin_url}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <LinkedInIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.github_username && (
+            <ListItemButton
+              component="a"
+              href={`https://github.com/${user.github_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <GitHubIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.website && (
+            <ListItemButton
+              component="a"
+              href={user.website}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <HomeRoundedIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+        </List>
+      </Container>
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<Props> = async ({ req, query }) => {
```

Comment:
can we make this use getInitialProps like we do in the [index page](https://github.com/timmyichen/dev-directory/blob/main/pages/index.tsx#L44)?  for consistency, and this way we save a round trip request to the server

Reply:
Even though it's not recommended? I was thinking we'd change Index to use `getServerSideProps` instead. 🤔

Reply:
yeah, in the [3/10 meeting](https://github.com/timmyichen/dev-directory/discussions/20) we decided to go with a REST API (which implicitly means we're using `getInitialProps`. `getServerSideProps` always runs on the server; next just makes a secret endpoint to fetch that data.  the flow would look like this:

getInitialProps:
* server side render: server -> makes request to endpoint -> renders -> sends to client
* client side render: client -> makes request to endpoint -> gets response

getServerSideProps:
* server side render: server -> makes request to endpoint -> renders -> sends to client
* client side render: client -> makes request to server -> server makes request to endpoint -> server receives data -> server passes data back to client

so we'd be making an extra fetch on the user side

the way that `getServerSideProps` is recommended is to do all the _direct_ data fetching from that function, rather than to call our own server's endpoint to get the data

[here](https://github.com/vercel/next.js/discussions/10930)'s a relevant thread if you want to give it a read!